backend/src/test/java/com/github/ingbeck/backend/service/DiaryServiceTest.java

```diff
@@ -75,4 +75,35 @@ void updateDiaryEntry_whenCalledWithExistingDiaryEntry_updateDiaryEntry(){
 
     }
 
+    @Test
+    void deleteFoodItem_whenCalledWithLastFoodItemInEntry_thenReturnNull(){
+        //GIVEN
+        String userId = "1";
+        String date = "2024-04-11";
+        FoodItem kinderriegel = new FoodItem("2","Kinderriegel", 1, "", 54, MealType.SNACK);
+        DiaryEntry existingDiaryEntry = new DiaryEntry(date, new ArrayList<>(List.of(kinderriegel)), 54);
+
+        when(diaryRepository.findDiaryByUserId(userId)).thenReturn(new Diary("1", "1", new ArrayList<>(List.of(existingDiaryEntry))));
+
+        //WHEN & THEN
+        diaryService.updateDiaryEntry(userId, date, kinderriegel);
+        verify(diaryRepository).save(any(Diary.class));
+    }
+
+    @Test
+    void deleteFoodItem_whenCalledMoreThen2FoodItemInEntryLeft_thenReturnUpdatedDiaryEntry(){
+        //GIVEN
+        String userId = "1";
+        String date = "2024-04-11";
+        FoodItem apfel = new FoodItem("1","Apfel", 50, "g", 140, MealType.SNACK);
+        FoodItem kinderriegel = new FoodItem("2","Kinderriegel", 1, "", 54, MealType.SNACK);
+        DiaryEntry existingDiaryEntry = new DiaryEntry(date, new ArrayList<>(List.of(apfel,kinderriegel)), 54);
+
+        when(diaryRepository.findDiaryByUserId(userId)).thenReturn(new Diary("1", "1", new ArrayList<>(List.of(existingDiaryEntry))));
+
+        //WHEN & THEN
+        diaryService.updateDiaryEntry(userId, date, kinderriegel);
```

Comment:
`assertNull(diaryService.updateDiaryEntry(userId, date, kinderriegel));` ? ^^

Reply:
Leider klappt das nicht, hatte ich auch versucht. 
In meiner deleteFoodItem Methode rufe ich getDiaryEntryByDate(userId, date) zwei mal auf. 
Damit ich am Ende eine assertion machen kann, müsste ich getDiaryEntryByDate(userId, date) zwei Mal in der selben Methode mocken. Damit hat Mockito ein gehöriges Problem :)